src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/design/sealedshouldhavepermitslist/InputSealedShouldHavePermitsListTopLevelSealedInterface.java

```diff
@@ -0,0 +1,19 @@
+/*
+SealedShouldHavePermitsList
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.design.sealedshouldhavepermitslist;
+
+public sealed interface InputSealedShouldHavePermitsListTopLevelSealedInterface { // violation
```

Comment:
Similar to above, let's spice these up with static classes, records, enums, etc

Reply:
I think it is done. Please review. Sorry for the late reply it was done  yesterday but I forgot to reply :)